Design Setup: Remove references to monthly subscriptions from upgrade modal

sixhours opened the pull request:
#### Proposed Changes

* We won't be offering themes for monthly subscriptions, so I've removed that information from the modal.

**Before**

<img width="817" alt="Screen Shot 2022-08-08 at 1 00 47 PM" src="https://user-images.githubusercontent.com/2124984/183473007-e08f0535-8780-4332-82c9-6690cecef536.png">




**After**

<img width="804" alt="Screen Shot 2022-08-08 at 12 53 05 PM" src="https://user-images.githubusercontent.com/2124984/183471900-9f44c280-4a6d-44b2-a6d5-c6a2abf1bec1.png">


#### Testing Instructions

* Switch to this PR, navigate to `/setup/?siteSlug=yourtestsite.wordpress.com&flags=signup/seller-upgrade-modal`
* Choose a Premium design from the design selection step
* Click "upgrade" in the upper right corner
* You should see the above modal instead of being forwarded to checkout.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

sixhours committed:
Remove references to monthly subscriptions from upgrade modal

mreishus reviewed (approved):
LGTM, this is an improvement.

---

After testing, I believe the modal needs some additional work in light of recent changes:

1. It's telling me the price of the pro plan, but clicking on the blue button puts the premium plan in my cart.

![2022-08-09_10-21](https://user-images.githubusercontent.com/937354/183690110-74da2c27-0cda-4fe5-a046-4836e4ce1d8e.png)
![2022-08-09_10-22](https://user-images.githubusercontent.com/937354/183690130-defa9a82-0ff9-4193-b5ea-23b9f9405683.png)

2. Because the button is one of two options: (Buy and activate theme) or (get it for free on the Premium plan), it makes it sound like the blue button is taking me to buy the theme individually, or to buy the premium themes addon. But it is actually taking me to buy the Premium plan.

sixhours commented:
> It's telling me the price of the pro plan, but clicking on the blue button puts the premium plan in my cart.

> Because the button is one of two options: (Buy and activate theme) or (get it for free on the Premium plan), it makes it sound like the blue button is taking me to buy the theme individually, or to buy the premium themes addon. But it is actually taking me to buy the Premium plan.

Yeah I had the pro plan in there as a placeholder since we don't have the theme's subscription products set up yet. I updated the plan reference to Premium for consistency with the cart behavior and added a todo to that effect. 

I think we're going to need design updates across the board to account for the behavior for free plan users vs. upgraded plan users...and you're right, the wording is somewhat repetitive and confusing!

sixhours committed:
Update plan placeholder to correct plan slug, add todo for updating to theme pricing

mreishus commented:
Seeing the price of the premium plan now :+1:
![2022-08-09_12-35](https://user-images.githubusercontent.com/937354/183721849-08eb55cd-f5c9-4840-8d72-3f1ab670643d.png)

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7469292

Thank you @sixhours for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7469292).
